main.py

```diff
@@ -53,6 +55,7 @@ def setup(self):
         self._intfs_tags_updated_at = {}
         self.link_up = set()
         self.link_status_lock = Lock()
+        self._switch_lock = Lock()
```

Comment:
The `Lock()` instance works here, but since this is a new introduced lock, and we actually need a lock per switch a `defaultdict(Lock)` would be more appropriate for this lock usage.